express/parsers/molecule.py

```diff
@@ -0,0 +1,111 @@
+from io import StringIO
+import os
+from ase.io import read, write
+import ase
+import rdkit
+from rdkit import Chem
+
+
+class MoleculeParser():
+    """
+    Molecule parser class.
+
+    Args:
+        args (list): args passed to the parser.
+        kwargs (dict): kwargs passed to the parser.
+            structure_string (str): structure string.
+            structure_format (str): structure format, poscar or espresso-in.
+    """
+
+    def __init__(self, structure_string):
+        self.structure_string = structure_string
+        self.inchi_run = self.try_openbabel_import()
+
+    def try_openbabel_import(self):
+        """
+        openbabel & pybel work together to generate the molecule string that
+        can then be converted into an InChI string by rdkit.
+
+        openbabel & pybel can be imported when the butler-venv is running.
+        Without the butler-venv, openbabel is not installed and therefore
+        both imports will fail.
+
+        When the butler-venv is not installed, inchi generation will be
+        set to 'None' and essentially skipped for the purpose of testing.
+
+        Without the 'try' statements ExPrESS will fail due to import errors.
+        If the openbabel import is successul, then pybel will be imported.
+        If the openbabel import fails, then get_inchi & get_inchi_key will
+        be set to 'None'
+
+        Returns:
+            Int:
+                0 = import failed, do not run InChI generation.
+                1 = import successful, run InChI generation.
+        """
+        try:
+            import pybel
```

Comment:
This import is not global, which is why i reimport pybel on line 64.

The benefit to putting the import at the top (as in the previous commit you reviewed) was that the import was global.

I think it's possible to make it a global import, but i'm not sure it's proper python coding. https://stackoverflow.com/questions/11990556/how-to-make-global-imports-from-a-function.

as is, with the second import on line 64 everything works.